### ISSUE TITLE
Configurable Products always out of stock

### ISSUE THREAD

**Issue by bareblends** [DivanteLtd/magento2-vsbridge-indexer#203]
Since upgrading to 1.9.0 configurable products' stock.is_in_stock is always equal to false, regardless of the stock status of child products (if stock changes in Model/Indexer/DataProvider/Product/ConfigurableData.php introduced in 1.9.0 are reverted then it works as expected)

**Comment by afirlejczyk** [comment]
Yes, line 291 should be
```
 if (empty($productDTO['stock']['is_in_stock']) || !$areChildInStock) {
```
instead of
```
 if (!empty($productDTO['stock']['is_in_stock']) || !$areChildInStock) {
```

**Comment by afirlejczyk** [comment]
Fixed with 1.9.1
